buildpacks/jvm-function-invoker/src/layers/opt.rs

```diff
@@ -0,0 +1,55 @@
+use std::fs;
+use std::os::unix::fs::PermissionsExt;
+use std::path::{Path, PathBuf};
+
+use libcnb::data::layer_content_metadata::LayerContentMetadata;
+use libcnb::layer_lifecycle::LayerLifecycle;
+use libcnb::{BuildContext, GenericMetadata, GenericPlatform};
+
+use crate::error::JvmFunctionInvokerBuildpackError;
+use crate::JvmFunctionInvokerBuildpackMetadata;
+
+pub struct OptLayerLifecycle {}
+
+#[cfg(target_family = "unix")]
+impl
+    LayerLifecycle<
+        GenericPlatform,
+        JvmFunctionInvokerBuildpackMetadata,
+        GenericMetadata,
+        PathBuf,
+        JvmFunctionInvokerBuildpackError,
+    > for OptLayerLifecycle
+{
+    fn create(
+        &self,
+        path: &Path,
+        context: &BuildContext<GenericPlatform, JvmFunctionInvokerBuildpackMetadata>,
+    ) -> Result<LayerContentMetadata<GenericMetadata>, JvmFunctionInvokerBuildpackError> {
+        let source = context.buildpack_dir.join("opt").join("run.sh");
+        let destination = path.join("run.sh");
+
+        fs::copy(&source, &destination).map_err(OptLayerError::CouldNotCopyRunSh)?;
+
+        fs::set_permissions(&destination, fs::Permissions::from_mode(0o755))
+            .map_err(OptLayerError::CouldNotSetExecutableBitForRunSh)?;
+
+        Ok(LayerContentMetadata::default().launch(true))
+    }
+
```

Comment:
There is no `on_lifecycle_start` here with log output. Is this intentional? (Quite possibility, given this is a simpler layer, but wanted to check given the other layers all output something.)

Reply:
Yeah, intentional for the exact reason you described :)